opacus/layers/dp_lstm.py

```diff
@@ -114,29 +175,60 @@ def forward(
             - ``c_n`` is of shape ``[B, H]`` tensor containing the cell state for ``t = T``.
         """
 
-        seq_length, batch_sz, _ = x.shape
-        if self.reverse:
-            x = x.flip(0)
-        x = torch.unbind(x, dim=0)
+        if batch_sizes is not None:
+            seq_length = batch_sizes.size(0)
+            if self.reverse:
+                x = tuple(reversed(x))
+                batch_sizes = batch_sizes.flip(0)
+        else:
+            seq_length, batch_sz, _ = x.shape
+            if self.reverse:
+                x = x.flip(0)
+            x = torch.unbind(x, dim=0)
 
         h_0, c_0 = state_init
 
         h_n = [h_0]
         c_n = [c_0]
+        batch_size_prev = h_0.shape[0]
 
         for t in range(seq_length):
-            h_next, c_next = self.cell(x[t], h_n[t], c_n[t])
+            if batch_sizes is not None:
```

Comment:
Let's handle a PackedSequence directly here, and let's just put this code into a helper function so that this stays readable